gpt4all-api/gpt4all_api/Dockerfile.buildkit

```diff
@@ -1,17 +1,28 @@
 # syntax=docker/dockerfile:1.0.0-experimental
 FROM tiangolo/uvicorn-gunicorn:python3.11
 
-# Put first so anytime this file changes other cached layers are invalidated.
+# Copy the requirements file to the image
 COPY gpt4all_api/requirements.txt /requirements.txt
 
-RUN pip install --upgrade pip
+# Set an argument for the model name, default value is 1000 but can be overridden or set to a different value
+ARG USER_ID
 
-# Run various pip install commands with ssh keys from host machine.
-RUN --mount=type=ssh pip install -r /requirements.txt && \
-  rm -Rf /root/.cache && rm -Rf /tmp/pip-install*
+# Create a non-root user 'appuser' and switch to it
+RUN useradd -u ${USER_ID} -m appuser 
```

Comment:
I tested this branch. I created a `.env` file by copying `env` with the default `USER_ID=1001`. Executing
`DOCKER_BUILDKIT=1 docker build -t gpt4all_api --progress plain -f gpt4all_api/Dockerfile.buildkit .` on ArchLinux resulted in the following error for me:

> #12 [3/6] RUN useradd -u  -m appuser
> #12 0.440 useradd: invalid user ID '-m'
> #12 ERROR: process "/bin/sh -c useradd -u ${USER_ID} -m appuser" did not complete successfully: exit code: 3

Am I doing something wrong?

Reply:
Ahh I see the problem, when building with docker build, you need to pass the user 1001 variable like this
`DOCKER_BUILDKIT=1 docker build --build-arg USER_ID=1001 -t gpt4all_api --progress plain -f gpt4all_api/Dockerfile.buildkit .
`

Will update the instructions, if you perform `make testenv` or `docker-compose -f docker-compose.yaml up --build` the variable will be passed from .env

Once I update the PR, you should be able to just run `DOCKER_BUILDKIT=1 docker build $(grep 'USER_ID' .env | xargs -I {} echo --build-arg {}) -t gpt4all_api --progress plain -f gpt4all_api/Dockerfile.buildkit .`

Reply:
Wait a minute, I did mention it on the new READ me :D

Reply:
> Wait a minute, I did mention it on the new READ me

Thanks, I overlooked that. It works :+1: 

I am now wondering: With "USER_ID=1001" passed to the `docker build` command, there are two sources for the `USER_ID`: 1) the `docker build` command and 2) the `.env` file. The `USER_ID` value might not necessarily be the same.

Do both `USER_ID`s have the same meaning? Does it makes sense to apply DRY here?

Reply:
Docker builds images using docker build and docker-compose. docker-compose automatically picks up environment variables, streamlining configuration management for multiple services. In contrast, docker build requires explicitly passing each environment variable as --build-arg, offering precise control over the build process but requiring more manual input.